core/scheduler/service/executor_input_compiler.go

```diff
@@ -62,6 +63,33 @@ type InputCompiler struct {
 	logger log.Logger
 }
 
+func sanitiseLabel(key string) string {
+	// implements validation from https://cloud.google.com/bigquery/docs/labels-intro#requirements
+
+	// max length 63 chars
+	// if length greater than 63 then truncate to 61 and add __ at end to denote truncation
+	if maxLabelLength := 63; len(key) > maxLabelLength {
+		key = key[len(key)-61:] + "__"
+	}
+	// Keys can contain only lowercase letters, numeric characters, underscores, and dashes.
+	key = strings.ToLower(key)
+	re := regexp.MustCompile(`[^\w-]`)
```

Comment:
Since , we are not creating regex from a Dynamic String, this is a deterministic action, if it compile once , it will always compile, so i think we are safe with it not being in the initialisation

This shall always compile successfully

Reply:
moved it to initialisation, to avoid redundant compilation on each invocation.